### PR TITLE
Feature single master secret, closing issue #78

### PR DESCRIPTION
**Pull request by gusmith**
Using key generation from secrets, we do not need to provide two secrets to create the clks as specified in #78 

The code has been updated, with all the tests and tutorials, and some documentation.
Do you think there would be a place where we were still using two secrets? I mainly wondering if I forgot some documentation.

### CODE REVIEW

**Comment by gusmith** [comment]
When using a deployed entity-service on my machine, all the tests pass, using the environment variables `INCLUDE_CLI` and `TEST_ENTITY_SERVICE`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@
 - Migrate to Azure DevOps for CI testing.
 - Synthetic data generation using distributions.
 
+BREAKING CHANGE:
+
+- The cli method `hash` requires only one secret instead of two.
```
Comment (wilko77) [inline CHANGELOG.md]: we should also mention that this change makes pre 0.14.0 clks incompatible to 0.14.0 clks!

File: clkhash/cli.py [hunk 2]
```diff
@@ -134,14 +134,14 @@ def cli(verbose):
 
 @cli.command('hash', short_help="generate hashes from local PII data")
 @click.argument('pii_csv', type=click.File('r'))
-@click.argument('keys', nargs=2, type=click.Tuple([str, str]))
+@click.argument('key', type=str, required=True)
```
[the remark this answers is not in the document]
Comment (gusmith) [reply clkhash/cli.py]: :+1:

File: CHANGELOG.md (hunk 1 above)
Comment (gusmith) [reply CHANGELOG.md]: Good point.

File: tests/test_key_derivation.py [hunk 3]
```diff
@@ -22,21 +22,20 @@ def test_kdf(self):
                     self.assertEqual(len(key), key_size)
 
     def test_generate_key_lists(self):
-        master_secrets = ['No, I am your father'.encode(),
-                          "No... that's not true! That's impossible!".encode()]
+        master_secret = "No, I am your fatherNo... that's not true! That's impossible!".encode()
```
[the remark this answers is not in the document]
Comment (gusmith) [reply tests/test_key_derivation.py]: :+1:

**Comment by gusmith** [comment]
@wilko77 all the changes you dreamed for are here, including the renaming which was indeed really confusing.
Just mentioning, the `force-push` was only to modify the last commit where I had a typo, so not a commit you had reviewed.

File: clkhash/key_derivation.py [hunk 4]
```diff
@@ -93,41 +96,45 @@ def generate_key_lists(master_secrets,  # type: Sequence[Union[bytes, str]]
...
-                keys.append(key)
-            else:
-                keys.append(key.encode('UTF-8'))
+        if isinstance(secret, bytes):
+            secret_bytes = secret
+        else:
+            secret_bytes = secret.encode('UTF-8')
     except AttributeError:
-        raise TypeError("provided 'master_secrets' have to be either of type bytes or strings.")
+        raise TypeError("provided 'secret' have to be either of type bytes or strings.")
```
Comment (wilko77) [inline clkhash/key_derivation.py]: typo: `has` instead of `have`

File: docs/tutorial_api.ipynb [hunk 5]
```diff
@@ -332,9 +332,9 @@
    "source": [
     "## Hash the data\n",
     "\n",
-    "We can now hash our PII data from the CSV file using our defined schema. We must provide a list of *secret keys* to this command - these keys have to be used by both parties hashing data. For this toy example we will use the keys _'key1'_ and _'key2'_, for real data, make sure that the keys contain enough entropy, as knowledge of these keys is sufficient to reconstruct the PII information from a CLK! \n",
+    "We can now hash our PII data from the CSV file using our defined schema. We must provide a *secret* to this command - this key has to be used by both parties hashing data. For this toy example we will use the secret _'secret'_, for real data, make sure that the key contains enough entropy, as knowledge of this key is sufficient to reconstruct the PII information from a CLK! \n",
```
Comment (wilko77) [inline docs/tutorial_api.ipynb]: You use `key` and `secret` for the same thing. Better stick with `secret`

File: docs/tutorial_api.ipynb [hunk 6]
```diff
@@ -332,9 +332,9 @@
...
     "\n",
-    "Also, **do not share these keys with anyone, except the other participating party.**"
+    "Also, **do not share this key with anyone, except the other participating party.**"
```
Comment (wilko77) [inline docs/tutorial_api.ipynb]: key -> secret

File: docs/tutorial_cli.ipynb [hunk 7]
```diff
@@ -405,7 +412,7 @@
    "source": [
     "## Hash the data\n",
     "\n",
-    "We can now hash our Personally Identifiable Information (PII) data from the CSV file using our defined linkage schema. We must provide two *secret keys* to this command - these keys have to be used by both parties hashing data. For this toy example we will use the keys _'key1'_ and _'key2'_, for real data, make sure that the keys contain enough entropy, as knowledge of these keys is sufficient to reconstruct the PII information from a CLK! Also, **do not share these keys with anyone, except the other participating party.**"
+    "We can now hash our Personally Identifiable Information (PII) data from the CSV file using our defined linkage schema. We must provide one *secret* to this command - this secret has to be used by both parties hashing data. For this toy example we will use the secret _'secret'_, for real data, make sure that the key contains enough entropy, as knowledge of this key is sufficient to reconstruct the PII information from a CLK! Also, **do not share this key with anyone, except the other participating party.**"
```
Comment (wilko77) [inline docs/tutorial_cli.ipynb]: again, don't use `key` here.

**Pull request merged**